### DISCUSSION TITLE
Validation Errors on Inferno Validator

### DISCUSSION THREAD

**Discussion by garyisaac** [synthetichealth/synthea#1400] [Q&A]
Hi,

I ran the attached file through the Inferno Validator report and it threw over 800 errors. It is validating against http://hl7.org/fhir/StructureDefinition/Bundle. Are patient files expected to pass this validation? Or, is there a configuration setting I need to enable? I generated the files using R4 and US Core 4 in the properties file.

[Inferno Resource Validator.pdf](https://github.com/synthetichealth/synthea/files/13514709/Inferno.Resource.Validator.pdf)
[8905245b-8bd8-a23c-0909-c6f08ed11f23.json](https://github.com/synthetichealth/synthea/files/13514717/8905245b-8bd8-a23c-0909-c6f08ed11f23.json)

**Comment by dehall** [comment]
In general, yes files produced by Synthea should pass validation without errors. Thanks for letting us know when they don't.
It looks like there are 2 things going on here, repeated over and over again for each resource in the Bundle: 
1. One error, repeated 200 times, is `Server Error: Wrong Display Name 'Evaluation+Plan note' for http://loinc.org#51847-2 - should be one of 32 choices: 'Evaluation + Plan note', 'Eval+Plan note', [...]` . In this case it looks like Synthea is using the wrong display value for the LOINC code, the spacing is wrong. This is something we should fix. Unfortunately these code systems change so display text that is valid today may become invalid later. (Also note that this type of mismatch used to be just a warning but turned into an error back in May)

2. The rest of the errors appear to be caused by validating this R4 resource against R5 definitions, for example there are a lot of errors related to the `Encounter.class` field: `4. Bundle.entry[1].resource.class: This property must be an Array, not an Object on line 178.`. This field changed from a 1..1 Coding in R4 to a 0..* CodeableConcept in R5, so expecting it to be an Array means it's expecting the R5 format.  I let the Inferno team know about this and they restarted the service, now when I upload the same resource there are just a few errors related to codes, which I'll take a look at separately. The Inferno team is looking into why this happened and how we can prevent it from happening again.

**Comment by garyisaac** [comment]
Thanks Dylan. I didn't know it is validating against R5.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Reply by garyisaac** [reply]
I appreciate the help Dylan. I made sure the validator is checking against R4 and it threw fewer errors. I think this is expected depending on what extensions are used in the Synthea file. Thanks again for your help.
